manifests/opsmanager/install.pp

```diff
@@ -0,0 +1,35 @@
+# PRIVATE CLASS: do not call directly
+class mongodb::opsmanager::install {
+  #assert_private("You are calling a private class mongodb::opsmanager::install.")
+  $package_ensure = $mongodb::opsmanager::package_ensure
+  $package_name   = $mongodb::opsmanager::package_name
+  $download_url   = $mongodb::opsmanager::download_url
+
+  case $package_ensure {
+    true:     {
+      $my_package_ensure = 'present'
+      $file_ensure       = 'directory'
+    }
+    false:    {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'absent': {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'purged': {
+      $my_package_ensure = 'purged'
+      $file_ensure       = 'absent'
+    }
+    default:  {
+      $my_package_ensure = $package_ensure
+      $file_ensure       = 'present'
+    }
+  }
+
+  package { $package_name:
+    ensure => $my_package_ensure,
+    source => $download_url,
```

Comment:
https://puppet.com/docs/puppet/6.0/type.html#package-attribute-source states:
> For example: the yum and apt providers ignore this attribute, but the rpm and dpkg providers require it.

So will this even work?

Reply:
Good spot. So you also have to override `provider` to `rpm` or `dpkg` based on `fact('os.family')`?

Reply:
It does download the package correctly, but it doesn't seem to ensure correctly. I've noticed that if anything besides mongodb-mms is used, when puppet runs for the second time it errors.

Reply:
Using the [package_provider fact](https://github.com/puppetlabs/puppetlabs-stdlib#package_provider) is probably cleaner than the os family fact.

Reply:
But specifically, you don't want the  'default provider' (which for RedHat, would be `yum`)

Reply:
So is the best solution to just do a check against `package_provider` and change it to `rpm` if the `os.family` is redhat?